.github/workflows/check_sp.yml

```diff
@@ -0,0 +1,76 @@
+name: Check Service Principal
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "35 6 * * *"
+
+jobs:
+  set-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      tests: ${{ steps.select-tests.outputs.tests }}
+    steps:
+      - name: Select Tests
+        id: select-tests
+        run: |
+          d="{'environment' :'dev'          , 'principle': 's165d01-keyvault-readonly-access'}"
+          t="{'environment' :'test'          , 'principle': 's165t01-keyvault-readonly-access'}"
+          pp="{'environment' :'preprod'          , 'principle': 's165t01-preprod-keyvault-readonly-access'}"
+          p="{'environment' :'prod'         , 'principle': 's165p01-keyvault-readonly-access'}"
+          tests="{ 'data':[ ${d} ,  ${t} ,  ${pp} ,  ${p} ]}"
+          echo "::set-output name=tests::${tests}"
+  check_expires:
+    name:      ${{matrix.data.environment}}/${{ matrix.data.principle }}
+    needs:     set-matrix
+    strategy:
+       matrix: ${{ fromJson(needs.set-matrix.outputs.tests) }}
+    environment:
+         name: ${{ matrix.data.environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check ${{ steps.serviceprinciple.outputs.name }}
+        uses: DFE-Digital/github-actions/CheckServicePrincipal@master
+        id: pwsh_check_expire
+        with:
+            AzureCredentials: ${{ secrets.AZURE_CREDENTIALS }}
+            ServicePrincipal: ${{ matrix.data.principle }}
+            ExpiresWithinDays: 30
+
+      - name: Processing ${{fromJson(steps.pwsh_check_expire.outputs.json_data).data.Name}}
+        run:  echo Expires in ${{fromJson(steps.pwsh_check_expire.outputs.json_data).data.ExpiresDays}} days
+
+      - name: Check out the repo
+        if:   fromJson(steps.pwsh_check_expire.outputs.json_data).data.Application != 'Dummy'
+        uses: actions/checkout@v2.4.0
+
+      - name: set-up-environment
```

Comment:
I believe this is only used to set up `SLACK_ERROR`. If it's only used in this workflow, we might as well simplify and hardcode the colour?